api/views/event_view.py

```diff
@@ -13,18 +13,52 @@
 )
 
 from .team_base import *
-from ..permissions import IsUserOwnerOrReadOnly
 
 class EventViewSet(TeamRelatedViewSet):
     queryset = Event.objects.all()
     serializer_class = EventSerializer
     permission_classes = [
         permissions.IsAuthenticated,
-        IsUserOwnerOrReadOnly | IsAdminMemberOrReadOnly,
+        IsAdminMemberOrReadOnly,
         IsMemberUser,
     ]
     filter_backends = [django_filters.rest_framework.DjangoFilterBackend]
     search_fields = ['^team__name', '^title', '^description', '^description', '^event_type',
                      '^user']
     ordering_fields = ['created', 'team', 'event_type', 'user']
-    
\ No newline at end of file
+
+    @action(methods=["GET"],
+            permission_classes=[permissions.IsAuthenticated, IsMemberUser]
+    )
+    def list(self, request, pk=None):
+        serializer = self.serializer_class(self.queryset, many=True)
+        event = get_object_or_404(Event, pk=pk)
+        self.check_object_permissions(request, event)
+        return Response(serializer.data, status=status.HTTP_200_OK)
+
+    @action(
+        detail=True,
+        methods=["GET"],
+        permission_classes=[permissions.IsAuthenticated, IsMemberUser]
+    )
+    def retrieve(self, request, *args, **kwargs):
+        instance = self.get_object()
+        serializer = self.get_serializer(instance)
+        response = serializer.data
+        return Response(response, status=status.HTTP_200_OK)
+
+    @action(
+        detail=True,
+        methods=["DELETE"],
+        permission_classes=[permissions.IsAuthenticated, IsMemberUser, IsAdminMemberOrReadOnly]
+    )
+    def destroy(self, request, *args, **kwargs):
```

Comment:
This looks good. Prefer to inherit from mixins.DeleteModelMixin and override that function, instead of registering your own url pattern. This is so that urls are mostly consistent along viewset.s

Reply:
do you mean DestroyModelMixin? I can't seem to find documentation on DeleteModelMixin. Also should I put that Mixin in the team related viewset or the event viewset?

Reply:
My bad, I meant the DestroyModelMixin. Put it in the Event model since not all team_related viewsets should have deletion